public/app/core/components/NestedFolderPicker/NestedFolderList.tsx

```diff
@@ -0,0 +1,99 @@
+import { css } from '@emotion/css';
+import React from 'react';
+import { FixedSizeList as List } from 'react-window';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { IconButton, useStyles2 } from '@grafana/ui';
+
+import { RootFolderWithUiState } from './types';
+
+const LIST_HEIGHT = 200;
+const ROW_HEIGHT = 40;
+
+interface RowProps {
+  index: number;
+  style: React.CSSProperties;
+  data: RootFolderWithUiState[];
+}
+
+function Row({ index, style, data }: RowProps) {
+  const folder = data[index];
+  const styles = useStyles2((theme) => getRowStyles(theme, folder.level));
```

Comment:
never ever pass a function to useStyles it makes it completely useless (it memoizes style creation) based on the function you pass to it

the styles should not be dynamically generated like this based on folder level. if you really need dynamic padding based on props use style prop. 

but if the html is nested should a single padding be enough?

Reply:
The HTML isn't nested, it all displays as a list, which is why the padding is needed for a "tree" structure

Reply:
@tskarhed then its best to just set style prop if just for dynamic padding of based on level.

Reply:
and keep emotion styles build once

Reply:
I'm taking over this now, but FYI for next time @tskarhed we have a component that does this for us

https://github.com/grafana/grafana/blob/c63a3796ffee9013f037d26e15465b7aecab9f7c/public/app/features/browse-dashboards/components/Indent.tsx#L12-L21